stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java

```diff
@@ -0,0 +1,186 @@
+        for (int i = 0; i < toAddressArrayLength; i++) {
+            toAddressArray[i] = new InternetAddress(toAddress[i]);
+        }
+        for (int i = 0; i < ccAddressArrayLength; i++) {
+            ccAddressArray[i] = new InternetAddress(ccAddress[i]);
+        }
+        for (int i = 0; i < bccAddressArrayLength; i++) {
+            bccAddressArray[i] = new InternetAddress(bccAddress[i]);
+        }
+        for (int i = 0; i < replyToAddressArrayLength; i++) {
+            replyToAddressArray[i] = new InternetAddress(replyToAddress[i]);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("[SmtpUtil][generateMessage] Generated TO, CC and BCC Address Arrays.");
+        }
+        String subject = message.getStringValue(EmailConstants.MESSAGE_SUBJECT);
+        String messageBody = message.getStringValue(EmailConstants.MESSAGE_MESSAGE_BODY);
+        String fromAddress = message.getStringValue(EmailConstants.MESSAGE_FROM);
+        if (fromAddress == null || fromAddress.isEmpty()) {
+            fromAddress = username;
+        }
+        String senderAddress = getNullCheckedString(message.getStringValue(EmailConstants.MESSAGE_SENDER));
+        if (log.isDebugEnabled()) {
+            log.debug("[SmtpUtil][generateMessage] Generating MimeMessage");
+        }
+
+        try {
+            emailMessage.setRecipients(Message.RecipientType.TO, toAddressArray);
+            if (ccAddressArrayLength > 0) {
+                emailMessage.setRecipients(Message.RecipientType.CC, ccAddressArray);
+            }
+            if (bccAddressArrayLength > 0) {
+                emailMessage.setRecipients(Message.RecipientType.BCC, bccAddressArray);
+            }
+            if (replyToAddressArrayLength > 0) {
+                emailMessage.setReplyTo(replyToAddressArray);
+            }
+            emailMessage.setSubject(subject);
+            emailMessage.setText(messageBody);
+            emailMessage.setFrom(new InternetAddress(fromAddress));
+            if (!senderAddress.isEmpty()) {
+                emailMessage.setSender(new InternetAddress(senderAddress));
+            }
+        } catch (MessagingException e) {
+            log.error("Failed to send message : ", e);
+        }
+        return emailMessage;
+    }
+
+    private static String[] getNullCheckedStringArray(MapValue mapValue, String parameter) {
+        ArrayValue arrayValue = mapValue.getArrayValue(parameter);
+        if (arrayValue != null) {
+            return arrayValue.getStringArray();
+        } else {
+            return new String[0];
+        }
+    }
+
+    private static int getNullArrayLengthChecked(String[] addresses) {
+        if (addresses == null) {
+            return 0;
+        } else {
+            return addresses.length;
+        }
+    }
+
+    private static String getNullCheckedString(String string) {
+        if (string != null) {
+            return string;
+        } else {
+            return "";
+        }
```

Comment:
We can use ternary operator and do this from a single line.

```suggestion
        return string != null ? string : "";
```

Reply:
Added to the commit.